test/sqllogictest/tpch.slt

```diff
@@ -1291,25 +1291,28 @@ WHERE
 
 %4 =
 | Get materialize.public.lineitem (u21)
-| ArrangeBy (#1)
+| Reduce group=(#1)
+| | agg sum(#4)
+| | agg count(true)
+| ArrangeBy (#0)
 
 %5 =
-| Join %3 %4 (= #0 #2)
+| Join %3 %4 (= #0 #1)
 | | implementation = DeltaQuery
-| |   delta %3 %4.(#1)
+| |   delta %3 %4.(#0)
 | |   delta %4 %3.(#0)
-| | demand = (#0, #5)
+| | demand = (#0, #2, #3)
 | Reduce group=(#0)
-| | agg sum(#5)
-| | agg count(true)
-| Map (2dec * (((#1 * 10000000dec) / i64todec(if (#2 = 0) then {null} else {#2})) / 10dec))
+| | agg sum(#2)
```

Comment:
The partial pushdown in TPCH query17 seems to be of marginal benefit. I did four trials of query 21 with 4 workers and logical compaction window 1ms.
before: 14985.760, 12949.214, 12298.098, 12373.173
after: 12064.085, 12591.950, 12880.420, 13784.646

Because this PR only has partial pushdowns, the outer reduce here is unnecessary. I think numbers will improve once full pushdown is enabled.